wallet/src/repository/wallet/mod.rs

```diff
@@ -108,12 +108,28 @@ where
                 // compute balance from utxo set if is not cached in the
                 // database, this is mostly used for testing where overflow
                 // checks are enabled
-                utxo_set
+                let (available, locked) = utxo_set
                     .iter()
-                    .map(|(_, balance)| balance.amount)
-                    .fold(0u64, |acc, amount| {
-                        acc.checked_add(amount).expect("balance overflow")
+                    .map(|(_, balance)| {
+                        (balance.amount, balance.time_lock)
                     })
+                    .fold((0u64, 0u64), |acc, (amount, time_lock)| {
```

Comment:
Here you can use a `BalanceInfo` instead of a tuple:

```suggestion
                    .fold(model::BalanceInfo { available: 0, locked: 0}, |acc, (amount, time_lock)| {
```